image/manifest.go

```diff
@@ -87,7 +87,22 @@ func (m *manifest) validate(w walker) error {
 	return nil
 }
 
-func (m *manifest) unpack(w walker, dest string) error {
+func (m *manifest) unpack(w walker, dest string) (retErr error) {
+	// error out if the dest directory is not empty
+	if _, err := os.Stat(dest); err == nil {
+		s, _ := ioutil.ReadDir(dest)
```

Comment:
You don't need both `Stat` and `ReadDir`.  If you want to error out if there is an entry at that path, just use `Stat`.  If you want to error out unless the path has no entry or contains an empty directory, just use `ReadDir`.  The latter sounds better to me (you can use it with `mktemp -d`), but I'm ok with either.